source/common/formatter/substitution_format_string.h

```diff
@@ -38,7 +39,8 @@ class SubstitutionFormatStringUtils {
       }
       auto typed_config = Envoy::Config::Utility::translateAnyToFactoryConfig(
           formatter.typed_config(), context.messageValidationVisitor(), *factory);
-      auto parser = factory->createCommandParserFromProto(*typed_config, context);
+      auto parser =
+          factory->createCommandParserFromProto(*typed_config, context.getServerFactoryContext());
```

Comment:
This bring potential logic change to third party forks (with very low possibility). Because scope(), initManager(), etc methods are different in the `FactoryContext` and `ServerFactoryContext`.

But lucky we never use it in our current formatter extensions.